src/twisted/trial/newsfragments/10230.bugfix

```diff
@@ -0,0 +1 @@
+trial.runner.filenameToModule now sets the correct module.__name__ and sys.modules key
```

Comment:
Minor comment...
I am not 100% I understand this release note.

```suggestion
trial.runner.filenameToModule now sets an absolute path to the module when updating  module.__name__ and sys.modules key
```

Reply:
This PR fixes the whole `module.__spec__` before the algorithm only got the `module.__file__` correct on 3.7- and 3.10+